src/mon/MDSMonitor.cc

```diff
@@ -144,7 +144,9 @@ void MDSMonitor::update_from_paxos(bool *need_bootstrap)
   // new map
   dout(4) << "new map" << dendl;
   print_map(fsmap, 0);
-  fsmap.sanity();
+  if (!g_conf->mon_mds_skip_sanity) {
+    fsmap.sanity();
+  }
```

Comment:
I decided that repeating it in MDSMonitor was the lesser of two evils compared with passing a reference to the config object through into FSMap